packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -0,0 +1,572 @@
+          ? ''
+          : widget.displayStringForOption(nextSelection);
+      _textEditingController.value = TextEditingValue(
+        selection: TextSelection.collapsed(offset: selectionString.length),
+        text: selectionString,
+      );
+      widget.onSelected?.call(_selection);
+    });
+  }
+
+  // Hide or show the options overlay, if needed.
+  void _updateOverlay() {
+    if (_shouldShowOptions) {
+      final RenderBox renderBox = context.findRenderObject() as RenderBox;
+      _floatingOptions?.remove();
+      _floatingOptions= OverlayEntry(
+        builder: (BuildContext context) {
+          return _FloatingOptions<T>(
+            optionsBuilder: widget.optionsBuilder,
+            fieldSize: renderBox.size,
+            layerLink: _optionsLayerLink,
+            onSelected: _select,
+            options: _options,
+          );
+        },
+      );
+      Overlay.of(context, rootOverlay: true)!.insert(_floatingOptions!);
+    } else if (_floatingOptions != null) {
+      _floatingOptions!.remove();
+      _floatingOptions = null;
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _textEditingController.addListener(_onChangedField);
+    SchedulerBinding.instance!.addPostFrameCallback((Duration _) {
+      _updateOverlay();
+    });
+  }
+
+  @override
+  void didUpdateWidget(AutocompleteCore<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    SchedulerBinding.instance!.addPostFrameCallback((Duration _) {
+      _updateOverlay();
+    });
+  }
+
+  @override
+  void dispose() {
+    _textEditingController.removeListener(_onChangedField);
+    _floatingOptions?.remove();
+    _floatingOptions = null;
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      key: _fieldKey,
+      child: CompositedTransformTarget(
+        link: _optionsLayerLink,
+        child: widget.fieldBuilder(
+          context,
+          _textEditingController,
+          _onFieldSubmitted,
+        ),
+      ),
+    );
+  }
+}
+
+// The floating options, meant to be built inside of an Overlay. Will position
+// itself at the bottom of the field indicated by layerLink and fieldSize.
+class _FloatingOptions<T> extends StatelessWidget {
+  const _FloatingOptions({
+    Key? key,
+    required this.optionsBuilder,
+    required this.fieldSize,
+    required this.layerLink,
+    required this.onSelected,
+    required this.options,
+  }) : assert(optionsBuilder != null),
+       assert(fieldSize != null),
+       assert(layerLink != null),
+       assert(onSelected != null),
+       assert(options != null),
+       super(key: key);
+
+  final AutocompleteOptionsBuilder<T> optionsBuilder;
+  final Size fieldSize;
+  final LayerLink layerLink;
+  final AutocompleteOnSelected<T> onSelected;
+  final List<T> options;
+
+  @override
+  Widget build(BuildContext context) {
+    return Positioned(
+      width: fieldSize.width,
+      child: CompositedTransformFollower(
```

Comment:
FYI I relanded the `CompositedTransformFollower` PR. But still I think it's probably not the best widget for positioning the results menu, as it gets cut off at screen edges.

Reply:
I was able to successfully use targetAnchor here, thank you for that!  (Though I do still need `fieldSize.width` like we discussed.)